tests/bluetooth/iso/modules/central.c

```diff
@@ -0,0 +1,490 @@
+LOG_MODULE_REGISTER(central, CONFIG_ISO_TEST_LOG_LEVEL);
+
+#define REMOTE_DEVICE_NAME_PEER	    CONFIG_BT_DEVICE_NAME
+#define REMOTE_DEVICE_NAME_PEER_LEN (sizeof(REMOTE_DEVICE_NAME_PEER) - 1)
+
+static bool running;
+static uint32_t acl_dummy_data_send_interval_ms = 100;
+static uint32_t acl_dummy_data_size = 20;
+struct peer_device {
+	struct bt_conn *conn;
+	struct k_work_delayable dummy_data_send_work;
+};
+static struct peer_device remote_peer[CONFIG_BT_MAX_CONN];
+static struct bt_nus_client *nus[CONFIG_BT_MAX_CONN];
+static struct bt_nus_client nus_client[CONFIG_BT_MAX_CONN];
+static struct bt_le_conn_param conn_param = {
+	.interval_min = 32, /* Set default ACL interval min to 40 ms */
+	.interval_max = 32, /* Set default ACL interval max to 40 ms */
+	.latency = 0,	    /* Set default slave latency to 0 */
+	.timeout = 40	    /* Set default supervision timeout to 4000 ms*/
+};
+
+static int scan_start(void);
+
+static bool all_peers_connected(void)
+{
+	for (int i = 0; i < ARRAY_SIZE(remote_peer); i++) {
+		if (remote_peer[i].conn == NULL) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static int channel_index_get(const struct bt_conn *conn, uint8_t *index)
+{
+	if (conn == NULL) {
+		LOG_ERR("No connection provided");
+		return -EINVAL;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(remote_peer); i++) {
+		if (remote_peer[i].conn == conn) {
+			*index = i;
+			return 0;
+		}
+	}
+
+	LOG_WRN("Connection not found");
+
+	return -EINVAL;
+}
+
+static void work_dummy_data_send(struct k_work *work)
+{
+	int ret;
+	char dummy_string[CONFIG_BT_MAX_CONN][CONFIG_BT_L2CAP_TX_MTU] = {0};
+	static uint8_t channel_index;
+	static uint32_t acl_send_count[CONFIG_BT_MAX_CONN];
+
+	struct peer_device *peer = CONTAINER_OF(work, struct peer_device, dummy_data_send_work);
+
+	ret = channel_index_get(peer->conn, &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+		return;
+	}
+	acl_send_count[channel_index]++;
+	sys_put_le32(acl_send_count[channel_index], dummy_string[channel_index]);
+
+	ret = bt_nus_client_send(&nus_client[channel_index], dummy_string[channel_index],
+				 acl_dummy_data_size);
+	if (ret) {
+		LOG_WRN("Failed to send, ret = %d", ret);
+	}
+
+	k_work_reschedule(&remote_peer[channel_index].dummy_data_send_work,
+			  K_MSEC(acl_dummy_data_send_interval_ms));
+}
+
+static void discovery_complete(struct bt_gatt_dm *dm, void *context)
+{
+	int ret;
+	uint8_t channel_index;
+
+	ret = channel_index_get(bt_gatt_dm_conn_get(dm), &channel_index);
+	if (ret) {
+		LOG_ERR("Channel index not found");
+	}
+
+	nus[channel_index] = context;
+
+	ret = bt_nus_handles_assign(dm, nus[channel_index]);
+	if (ret) {
+		LOG_ERR("NUS handle assigne failed, ret = %d", ret);
+	}
+	bt_nus_subscribe_receive(nus[channel_index]);
+
+	ret = bt_gatt_dm_data_release(dm);
+	if (ret) {
+		LOG_ERR("Release service discovery data failed, ret = %d", ret);
+	}
+
+	k_work_schedule(&remote_peer[channel_index].dummy_data_send_work, K_MSEC(100));
```

Comment:
We need to discuss this delay.

Reply:
This is the time between "service discover finished" and "start sending dummy data over ACL", we can have a quick discuss on Monday.